src/Rector/Namespace_/FactoryDefinitionRector.php

```diff
@@ -0,0 +1,228 @@
+use Rector\Laravel\NodeFactory\ModelFactoryNodeFactory;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://laravel.com/docs/7.x/database-testing#writing-factories
+ * @see https://laravel.com/docs/8.x/database-testing#defining-model-factories
+ *
+ * @see \Rector\Laravel\Tests\Rector\Namespace_\FactoryDefinitionRector\FactoryDefinitionRectorTest
+ */
+final class FactoryDefinitionRector extends AbstractRector
+{
+    public function __construct(
+        private ModelFactoryNodeFactory $modelFactoryNodeFactory
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Upgrade legacy factories to support classes.', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Faker\Generator as Faker;
+
+$factory->define(App\User::class, function (Faker $faker) {
+    return [
+        'name' => $faker->name,
+        'email' => $faker->unique()->safeEmail,
+    ];
+});
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+use Faker\Generator as Faker;
+
+class UserFactory extends \Illuminate\Database\Eloquent\Factories\Factory
+{
+    protected $model = App\User::class;
+    public function definition()
+    {
+        return [
+            'name' => $this->faker->name,
+            'email' => $this->faker->unique()->safeEmail,
+        ];
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Namespace_::class, FileWithoutNamespace::class];
+    }
+
+    /**
+     * @param Namespace_|FileWithoutNamespace $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $factories = [];
+        foreach ($node->stmts as $key => $stmt) {
+            if (! $stmt instanceof Expression) {
+                continue;
+            }
+
+            // do not refactor if factory variable changed
+            if ($this->isReAssignFactoryVariable($stmt->expr)) {
+                return null;
+            }
+
+            if (! $stmt->expr instanceof MethodCall) {
+                continue;
+            }
+
+            if ($this->shouldSkipExpression($stmt->expr)) {
+                continue;
+            }
+
+            $name = $this->getNameFromClassConstFetch($stmt->expr->args[0]->value);
+            if (! $name instanceof Name) {
+                continue;
+            }
+
+            if (! isset($factories[$name->toString()])) {
+                $factories[$name->toString()] = $this->createFactory($name->getLast(), $stmt->expr->args[0]->value);
+            }
+
+            $this->processFactoryConfiguration($factories[$name->toString()], $stmt->expr);
+
+            unset($node->stmts[$key]);
+        }
+
+        foreach ($factories as $factory) {
+            $node->stmts[] = $factory;
+        }
+
+        return $node;
```

Comment:
should return null when `$factories` is empty: 

```suggestion
        if ($factories === []) {
            return null;
        }

        $node->stmts = array_merge($node->stmts, $factories);
        return $node;
```